docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md

```diff
@@ -1,62 +1,108 @@
 
 - They didn't complete the read operation until either the buffer passed to the read operation was completely filled or the end of the stream was reached.
 - As wrapper streams, they didn't delegate zero-length buffer functionality to the stream they wrap.
 
-This change addresses both of these issues.
+Consider this example that creates and compresses 150 random bytes. It then sends the compressed data one byte at a time from the client to the server, and the server decompresses the data by calling `Read` and requesting all 150 bytes.
+
+:::code language="csharp" source="./snippets/ReadStream.cs":::
+
+In previous versions of .NET and .NET Framework, the following output shows that `Read` was only called once. Even though data was available for `GZipStream` to return, `Read` was forced to wait until the requested number of bytes was available.
+
+```output
+Read: 150 bytes
+Total received: 150 bytes
+```
+
+In .NET 6 and later versions, the following output shows that `Read` was called multiple times until *all* the requested data was received. Even though the call to `Read` requests 150 bytes, each call to `Read` was able to successfully decompress *some* bytes (that is, all of the bytes that had been received at that time) to return, and it did:
+
+```output
+Read: 1 bytes
+Read: 101 bytes
+Read: 4 bytes
+Read: 4 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 3 bytes
+Read: 2 bytes
+Read: 3 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Total received: 150 bytes
+```
 
 ## Old behavior
 
-When `Stream.Read` or `Stream.ReadAsync` was called on one of the affected stream types with a buffer of length `N`, the operation would not complete until:
+When `Stream.Read` or `Stream.ReadAsync` was called on one of the affected stream types with a buffer of length `N`, the operation wouldn't complete until:
 
 - `N` bytes had been read from the stream, or
-- The underlying stream they wrap returned 0 from a call to its read, indicating no more data is available.
+- The underlying stream they wrap returned 0 from a call to its read, indicating no more data was available.
 
 Also, when `Stream.Read` or `Stream.ReadAsync` was called with a buffer of length 0, the operation would succeed immediately, sometimes without doing a zero-length read on the stream it wraps.
 
 ## New behavior
 
 Starting in .NET 6, when `Stream.Read` or `Stream.ReadAsync` is called on one of the affected stream types with a buffer of length `N`, the operation completes when:
 
-- At least one byte has been read from the stream, or
+- At least 1 byte has been read from the stream, or
 - The underlying stream they wrap returns 0 from a call to its read, indicating no more data is available.
```

Comment:
This one doesn't have the same changes as made in the **old behavior** section.

Reply:
Do you mean there's no behavior difference between old/new?